tcases-lib/src/main/java/org/cornutum/tcases/io/AbstractSystemInputWriter.java

```diff
@@ -0,0 +1,107 @@
+//////////////////////////////////////////////////////////////////////////////
+// 
+//                    Copyright 2018, Cornutum Project
+//                             www.cornutum.org
+//
+//////////////////////////////////////////////////////////////////////////////
+
+package org.cornutum.tcases.io;
+
+import org.cornutum.tcases.SystemInputDef;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+
+/**
+ * Base class for writing a {@link SystemInputDef} to an output stream.
+ *
+ */
+public abstract class AbstractSystemInputWriter implements Closeable
+  {
+  /**
+   * Creates a new AbstractSystemInputWriter object that writes to standard output.
+   */
+  public AbstractSystemInputWriter()
+    {
+    this( System.out);
```

Comment:
Does this constructor provide enough value? To me it seems flawed to mention System.out anywhere in library code (unless the library is a logging library). Clients can pass System.out easily with the other constructors where needed.

Reply:
Hm, I guess it's for consistency with the previous code.

Reply:
It's based on the principle that a class should have a meaningful default constructor.